feat: Adopt uv for local development & production docker build

kbumsik opened the pull request:
Advantage is described in https://deepauto.slack.com/archives/C08CXL9HGUE/p1741583110510219

gmlwns2000 committed:
update BIG

gmlwns2000 committed:
fix bug

gmlwns2000 committed:
Merge pull request #28 from DeepAuto-AI/feat/update-example

Backport of #27

kbumsik committed:
doc: Update r1 qwen 32b usage

kbumsik committed:
feat: Update SGLang commit and update Docker to use CUDA 12.8

kbumsik committed:
Merge pull request #29 from DeepAuto-AI/feat/pre-prod

Feat/pre prod

kbumsik committed:
Merge pull request #30 from DeepAuto-AI/feat/research-diag

Release:2025-02-07

daniel-geon-park committed:
change sglang branch in dependencies

daniel-geon-park committed:
Merge branch 'feat/release-cand' into deepauto/release

kbumsik committed:
feat(uv): Initial uv sync

kbumsik committed:
feat(uv): Add flashinfer index

kbumsik committed:
feat(uv): Install hip-attn[all,no-build-iso] by default

kbumsik committed:
feat(uv): Update .gitignore to ignore .venv

kbumsik committed:
feat(uv): Fix python version to 3.11

kbumsik committed:
feat(uv): First successful clean build

kbumsik committed:
feat(uv): Simplify flash-attn process

kbumsik committed:
feat(docker): Build docker using uv

kbumsik committed:
doc(uv): Update USAGE.sglang with

kbumsik commented on README.md:
```diff
@@ -63,7 +63,28 @@ print(output.shape)
+cd hip-attention
+
+# This install all dev dependencies in .venv/
+uv sync
```
Now just a single `uv sync` command will install everything, including sglang and research

kbumsik commented on uv.lock:
The lock file will ensure all researchers have the same packages.

gmlwns2000 committed:
add wandb as research

gmlwns2000 reviewed (approved):
It works well with my environment!

kbumsik committed:
feat(uv) set sglang source

kbumsik committed:
feat(uv): Add not on how to locally develop

kbumsik committed:
feat(uv): Cleanup sglang section

kbumsik commented on pyproject.toml:
```diff
@@ -61,21 +64,49 @@ test = [  # Packages required for running the tests
+    # pip format: "sglang[all] @ git+https://github.com/DeepAuto-AI/sglang.git@deepauto/release#subdirectory=python"
+    { git = "https://github.com/DeepAuto-AI/sglang.git", subdirectory = "python", rev = "deepauto/release" },
+    # For local devs, you can use this instead:
+    # { path = "../sglang/python", editable = true },
```
How to link local sglang folder

daniel-geon-park reviewed (commented):
I'm new to uv, but is it still possible to use pip to install this package even after this merge? I think some people are still more comfortable with using pip to install packages.

kbumsik commented:
> I'm new to uv, but is it still possible to use pip to install this package even after this merge? I think some people are still more comfortable with using pip to install packages.

pyproject.toml is a standardized package specification, and both uv and pip follows the spec. Meaning that the section

```toml
[build-system]
requires = ["setuptools", "setuptools-scm"]
requires = [
   # anything
]
build-backend = "setuptools.build_meta"
```
tells both uv and pip to actually use `setuptools` to build the package, and `[tool.setuptools]` will be also respected. So the building process remains the same.

Currently the difference is that I removed git location in `project.dependencies` section like `sglang @ git+https://github.com/DeepAuto-AI/sglang.git@deepauto/dev#subdirectory=python` and moved to `tools.uv` section. But we need to remove the git location in `project.dependencies` section because PyPi will reject custom package location (for security) when we publish the package anyway.

So there will be a problem when installing `pip install .[sglang]` because it will point to the the upstream version. I will update README.md regarding that.

kbumsik commented:
So in a short, a compliant package manager (uv and pip) should follow pyproject.toml fields other than `[tool.*]` section. Package manager-specific configs go there.

kbumsik committed:
doc(pip): Update pip after introduing uv

kbumsik commented:
Updated README for pip

daniel-geon-park reviewed (approved):
Looks good to me!

kbumsik commented:
Thanks!
